File: animal_speak.py
'''
Making an animal speak using Classes and Inheritance
'''

class Animal:
    def __init__(self, n):
        self.name = n
    
    def speak(self):
        return f'{self.name} says hello!'

class Dog(Animal):
    def speak(self):
        return f'{self.name} barks!'

class Cat(Animal):
    def speak(self):
        return f'{self.name} meows!'

File: animal_speak.rb
class Animal
  def initialize(name)
    @name = name
  end

  def speak
    puts "#{@name} says hello!"
  end
end

class Dog < Animal
  def speak
    puts "#{@name} barks!"
  end
end

class Cat < Animal
  def speak
    puts "#{@name} meows!"
  end
end
